File: app/src/main/java/com/blackmirror/dongda/ui/activity/apply/ApplyContract.java
package com.blackmirror.dongda.ui.activity.apply;


import com.blackmirror.dongda.domain.model.ApplyServiceDomainBean;
import com.blackmirror.dongda.domain.model.BaseDataBean;
import com.blackmirror.dongda.ui.BasePresenter;

public class ApplyContract {
    public interface View{
        void onApplySuccess(ApplyServiceDomainBean bean);

        void onError(BaseDataBean bean);

    }

    public interface Presenter extends BasePresenter<View> {
        void apply(String brand_name, String name, String category, String phone, String city);
    }
}

File: app/src/main/java/com/blackmirror/dongda/ui/activity/apply/ApplyContract.kt
package com.blackmirror.dongda.ui.activity.apply


import com.blackmirror.dongda.kdomain.model.ApplyServiceDomainBean
import com.blackmirror.dongda.kdomain.model.BaseDataBean
import com.blackmirror.dongda.ui.BasePresenter

class ApplyContract {
    interface View {
        fun onApplySuccess(bean: ApplyServiceDomainBean)

        fun onError(bean: BaseDataBean)

    }

    interface Presenter : BasePresenter<View> {
        fun apply(brand_name: String, name: String, category: String, phone: String, city: String)
    }
}
